docs/bazel/README.md

```diff
@@ -0,0 +1,211 @@
+# Building with Bazel
+
+**ToC**
+* [Prerequisites](#prerequisites)
+* [Getting Started](#getting-started)
+* [Development](development.md)
+* [Status](status.md)
+
+## Prerequisites
+
+The current implementation is not hermetic - it relies on the host
+development environment.  For example, it reads the environment
+variables OPAMROOT and OPAM_SWITCH_PREFIX; if these are not set you
+will get an error message telling you what to do.  The final version will be hermetic, meaning it will automatically install all dependencies.
```

Comment:
In principle, Bazel rules could completely replace opam (and ocamlfind) as well as dune.  But we can also work with them. The current rules use ocamlfind, for example.  I expect the "final" version to use opam, with the user supplying a list of packages to install. It should also support the case where the user has not installed opam. This could be hermetic (pkgs installed internally) or not (bazel uses pkgs installed on host). Either way the user would be relieved of the need to run opam.

Reply:
That makes sense. As long as it's possible to use an existing opam install, I think I'm happy.

I'm interested to see how the hermetic install will look and work, but there's no urgency there, since I don't think you're working on that part currently.

Reply:
You might take a look at [jin/rules_ocaml]( https://github.com/jin/rules_ocaml) which does something like that in repo.bzl.  It's pretty simple, just runs  "opam install" on the list of libs as part of bootstrapping:

```
[repository_ctx.execute([
      opam_bin,
      "install",
      "%s=%s" % (pkg, version),
      "--yes",
      "--root", OPAM_ROOT_DIR
  ], quiet = DEBUG_QUIET)
   for (pkg, version) in repository_ctx.attr.opam_packages.items()]
```

Reply:
Do you have plans for checking version constraints with opam (e.g. bail if the version of `core` is too new)? We don't have this now, but this has tripped quite a few people up, so it would be a win if it's not too hard to add.

Reply:
Hmm, for Coda I suppose we'll list the deps and version numbers as config constants, so the user won't have to deal with them.  A general mechanism for dealing with versioning would be needed for OBazl, but I think that can wait.  If we wanted to get fancy we could support something like [Dart version constraints](https://dart.dev/tools/pub/versioning#version-constraints-the-dart-approach). I'm not so eager to write that code atm, but maybe we can find something like that in another Starlark library.